src/vsphere_cpi/lib/cloud/vsphere/cloud.rb

```diff
@@ -697,23 +697,22 @@ def generate_network_env(devices, networks, dvs_index)
       network_env
     end
 
-    def generate_disk_env(system_disk, ephemeral_disk, vm_config)
+    def generate_disk_env(system_disk, ephemeral_disk, disk_uuid_enabled)
 
       # When disk.enableUUID is true on the vmx options, consistent volume IDs are requested, and we can use them
       # to ensure the precise ephemeral volume is mounted.   This is mandatory for
       # cases where multiple SCSI controllers are present on the VM, as is common with Kubernetes VMs.
-
-      if vm_config.vmx_options['disk.enableUUID'] == "1"
+      if disk_uuid_enabled
         {
           'system' => system_disk.unit_number.to_s,
```

Comment:
@svrc-pivotal : I am trying to understand the reason behind retaining the unit number for the system disk. Is it irrelevant to the issue or is it required?

Reply:
Good question @eleanorigby .. the bosh agent requires it to be a string value, either the /dev/sdX device or unit number.  It will fail parsing if it’s a JSON string containing the uuid.

Reply:
@svrc-pivotal : Got it. It should continue to work then